### ISSUE TITLE
Minimum Cython version in documentation incorrect

### ISSUE THREAD

**Issue by jjhelmus** [h5py/h5py#587]
The minimum version of Cython required to build h5py from source is incorrect in the documentation and setup.py file.  These both state Cython version 0.17 or later yet using a version of Cython lower than 0.19 cannot cythonize the `/h5py/_conv.pyx` file.  Specifying Cython version 0.19 or later seems reasonable.

Partial log when building with Cython 0.18:

```
$ python setup.py build
zip_safe flag not set; analyzing archive contents...
...
Autodetected HDF5 1.8.5
********************************************************************************
                       Summary of the h5py configuration

    Path to HDF5: None
    HDF5 Version: '1.8.5'
     MPI Enabled: False
Rebuild Required: False

********************************************************************************
Executing api_gen rebuild of defs
Executing cythonize()
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/defs.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/_errors.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/_objects.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/_proxy.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5fd.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5z.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5i.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5r.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/utils.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/_conv.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5t.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5s.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5p.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5d.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5a.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5f.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5g.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5l.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5o.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5ds.pyx because it changed.
Compiling /home/jhelmus/python/h5py-2.5.0/h5py/h5ac.pyx because it changed.
Cythonizing /home/jhelmus/python/h5py-2.5.0/h5py/_conv.pyx

Error compiling Cython file:
------------------------------------------------------------
...
cdef herr_t H5Pset_fapl_family( hid_t fapl_id,  hsize_t memb_size, hid_t memb_fapl_id ) except *
cdef herr_t H5Pget_fapl_family( hid_t fapl_id, hsize_t *memb_size, hid_t *memb_fapl_id ) except *
cdef herr_t H5Pset_family_offset( hid_t fapl_id, hsize_t offset) except *
cdef herr_t H5Pget_family_offset( hid_t fapl_id, hsize_t *offset) except *
cdef herr_t H5Pset_fapl_log(hid_t fapl_id, char *logfile, unsigned int flags, size_t buf_size) except *
cdef herr_t H5Pset_fapl_multi(hid_t fapl_id, H5FD_mem_t *memb_map, hid_t *memb_fapl, const char * const *memb_name, haddr_t *memb_addr, hbool_t relax) except *
                                                                                                       ^
------------------------------------------------------------

h5py/defs.pxd:178:104: Expected ')', found '*'
```

**Comment by andrewcollette** [comment]
Looks like versions before 0.19 don't understand "const".  Thanks for letting us know!

**Comment by pdebuyl** [comment]
It could help to display explicitly the Cython version in the Travis builds. This way, we would now at least what is currently working on an automated basis.

**Comment by aragilar** [comment]
Closed by #650.
